types/utils.d.ts

```diff
@@ -31,7 +31,10 @@ export type ReactChildren = React.ReactNodeArray | React.ReactNode
 export type ReactPropsStrict<T> = { [K in keyof T]: NullableIfUndefined<T[K]> }
 export type ReactProps<T> = Extendable<ReactPropsStrict<T>>
 
-export type ComponentEventHandler<TProps> = (event: React.SyntheticEvent, data: TProps) => void
+export type ComponentEventHandler<TProps> = (
```

Comment:
```suggestion
export type ComponentEventHandler<TProps, TElem = HTMLElement> = (
```

Can we make this a generic?

Reply:
yes, although not sure whether it is necessary to introduce this variability now (as it is pretty much clear that it will always be about HTMLElement for this handler's type)